File: win-prog/ProgWinVB/Text and Fonts/HowdyWorldFullFit/HowdyWorldFullFit.vb
'--------------------------------------------------
' HowdyWorldFullFit.vb (c) 2002 by Charles Petzold
'--------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class HowdyWorldFullFit
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New HowdyWorldFullFit())
    End Sub

    Sub New()
        Text = "Howdy, world!"
        MinimumSize = _
                Size.op_Addition(SystemInformation.MinimumWindowSize, _
                                 New Size(0, 1))
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim fnt As New Font("Times New Roman", 10, FontStyle.Italic)
        Dim szf As SizeF = grfx.MeasureString(Text, fnt)
        Dim fScaleHorz As Single = cx / szf.Width
        Dim fScaleVert As Single = cy / szf.Height

        grfx.ScaleTransform(fScaleHorz, fScaleVert)
        grfx.DrawString(Text, fnt, New SolidBrush(clr), 0, 0)
    End Sub
End Class

File: win-prog/ProgWinCS/Text and Fonts/HowdyWorldFullFit/HowdyWorldFullFit.cs
//------------------------------------------------
// HowdyWorldFullFit.cs © 2001 by Charles Petzold
//------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class HowdyWorldFullFit: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new HowdyWorldFullFit());
     }
     public HowdyWorldFullFit()
     {
          Text = "Howdy, world!";
          MinimumSize = SystemInformation.MinimumWindowSize + new Size(0,1);
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Font  font  = new Font("Times New Roman", 10, FontStyle.Italic);
          SizeF sizef = grfx.MeasureString(Text, font);
          float fScaleHorz = cx / sizef.Width;
          float fScaleVert = cy / sizef.Height;

          grfx.ScaleTransform(fScaleHorz, fScaleVert);

          grfx.DrawString(Text, font, new SolidBrush(clr), 0, 0);
     }
}
